executor/aggfuncs/func_group_concat.go

```diff
@@ -321,14 +341,14 @@ func (h *topNRows) tryToAdd(row sortRow) (truncated bool, sortRowMemSize int64)
 			h.rows[0].buffer.Truncate(h.rows[0].buffer.Len() - int(debt))
 		} else {
 			h.currSize -= uint64(h.rows[0].buffer.Len()) + h.sepSize
+			memDelta -= int64(h.rows[0].buffer.Cap())
```

Comment:
Why do we need to `-Cap` here?

Reply:
The element has been pop from the heap. So the memory usage of the row should be minus.